packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -16,30 +18,30 @@ export enum OrgType {
 export async function getWorkspaceOrgType(
   defaultUsernameOrAlias?: string
 ): Promise<OrgType> {
-  if (!defaultUsernameOrAlias) {
-    const e = new Error();
-    e.name = 'NoDefaultusernameSet';
-    throw e;
-  }
-  const username = await OrgAuthInfo.getUsername(defaultUsernameOrAlias);
-  const isScratchOrg = await OrgAuthInfo.isAScratchOrg(username).catch(err =>
-    telemetryService.sendException(
-      'get_workspace_org_type_scratch_org',
-      err.message
-    )
-  );
-  return isScratchOrg ? OrgType.SourceTracked : OrgType.NonSourceTracked;
+  const connection = await WorkspaceContext.getInstance().getConnection();
+  const org: Org = await Org.create({ connection });
+  // Org.supportsSourceTracking() checks the org at this point in time.  This
+  // is important because a sandbox can be created without source tracking and
+  // then be refreshed to have source tracking.  So, in the case of sandboxes,
+  // we have to check the org's source tracking status in real time and can't
+  // rely on Org.tracksSource() because that property is populated at authorization-
+  // time only and is not updated after the Sandbox is refreshed unless the User
+  // re-authenticates that sandbox org.
+  // Also opting to use this call only and specifically NOT use org.isScratch(),
+  // which was not returning consistent results during dev.
+  const isSourceTracked = await org.supportsSourceTracking();
+  return isSourceTracked ? OrgType.SourceTracked : OrgType.NonSourceTracked;
 }
 
 export function setWorkspaceOrgTypeWithOrgType(orgType: OrgType) {
   setDefaultUsernameHasChangeTracking(orgType === OrgType.SourceTracked);
-  setDefaultUsernameHasNoChangeTracking(orgType === OrgType.NonSourceTracked);
+  setDefaultUsernameHasNoChangeTracking(orgType !== OrgType.SourceTracked);
 }
 
 export async function setupWorkspaceOrgType(defaultUsernameOrAlias?: string) {
   try {
     setHasDefaultUsername(!!defaultUsernameOrAlias);
-    const orgType = await getWorkspaceOrgType(defaultUsernameOrAlias);
```

Comment:
This might be out of scope but since you're touching this file... does anything actually use the getDefaultUsernameOrAlias? I'm wondering if we could remove that from this file and the vscode-integration test.ts to continue that cleanup too.

Reply:
Good call.  Removed!